tests/baselines/reference/destructuringTuple.errors.txt

```diff
@@ -25,22 +15,9 @@ destructuringTuple.ts(11,60): error TS2769: No overload matches this call.
     
     const [oops1] = [1, 2, 3].reduce((accu, el) => accu.concat(el), []);
           ~~~~~~~
-!!! error TS2461: Type 'number' is not an array type.
-                                                   ~~~~~~~~~~~~~~~
-!!! error TS2769: No overload matches this call.
-!!! error TS2769:   Overload 1 of 3, '(callbackfn: (previousValue: number, currentValue: number, currentIndex: number, array: number[]) => number, initialValue: number): number', gave the following error.
-!!! error TS2769:     Type 'never[]' is not assignable to type 'number'.
-!!! error TS2769:   Overload 2 of 3, '(callbackfn: (previousValue: [], currentValue: number, currentIndex: number, array: number[]) => [], initialValue: []): []', gave the following error.
-!!! error TS2769:     Type 'never[]' is not assignable to type '[]'.
-!!! error TS2769:       Target allows only 0 element(s) but source may have more.
-!!! related TS6502 lib.es5.d.ts:--:--: The expected type comes from the return type of this signature.
-!!! related TS6502 lib.es5.d.ts:--:--: The expected type comes from the return type of this signature.
-                                                               ~~
-!!! error TS2769: No overload matches this call.
-!!! error TS2769:   Overload 1 of 2, '(...items: ConcatArray<never>[]): never[]', gave the following error.
-!!! error TS2769:     Argument of type 'number' is not assignable to parameter of type 'ConcatArray<never>'.
-!!! error TS2769:   Overload 2 of 2, '(...items: ConcatArray<never>[]): never[]', gave the following error.
-!!! error TS2769:     Argument of type 'number' is not assignable to parameter of type 'ConcatArray<never>'.
+!!! error TS2461: Type 'unknown' is not an array type.
+                                                   ~~~~
+!!! error TS18046: 'accu' is of type 'unknown'.

```

Comment:
I'm wishy washy on this one. Previously, we'd fix the first parameter to `[]` at the end of the first overload's checking, and then witness it while checking the second overload and that'd produce these errors. Now, since both overloads fail, we fall back to the constraint on the type parameter (`unknown`), which produces different errors.